src/BaseListView.php

```diff
@@ -347,6 +347,11 @@ private function prepareDataReaderByParams(
     ): ReadableDataInterface {
         $dataReader = $this->getDataReader();
 
+        if ($dataReader instanceof LimitableDataInterface && $dataReader->getLimit() !== null) {
```

Comment:
Should we do it for grid only?

Reply:
I think, it is actual for all `BaseListView` implementations.